File: benchmarks/rust-runner/src/benchmark_row.rs
//! Java/Rust 性能测试共用的四列表格模型。

use chrono::{Duration, NaiveDate};
use easyexcel::ExcelRow;

/// 对应 Java：无直接对应对象；Rust 架构扩展。 与 Java `BenchmarkRow` 语义一致的性能测试行。
#[derive(Debug, Clone, ExcelRow)]
pub(crate) struct BenchmarkRow {
    #[excel(name = "ID", index = 0)]
    pub(crate) id: i64,
    #[excel(name = "Name", index = 1)]
    pub(crate) name: String,
    #[excel(name = "Date", index = 2)]
    pub(crate) date: NaiveDate,
    #[excel(name = "Score", index = 3)]
    pub(crate) score: f64,
}

impl BenchmarkRow {
    /// 对应 Java：无直接对应对象；Rust 架构扩展。 按统一契约生成第 `id` 行。
    #[allow(clippy::cast_precision_loss)]
    pub(crate) fn from_id(id: i64) -> Self {
        let base = NaiveDate::from_ymd_opt(2024, 1, 1).expect("valid benchmark epoch");
        Self {
            id,
            name: format!("row-{id}"),
            date: base + Duration::days(id.rem_euclid(28)),
            score: id as f64 * 0.5,
        }
    }
}

File: benchmarks/java-runner/src/com/alibaba/easyexcel/test/benchmark/BenchmarkRow.java
package com.alibaba.easyexcel.test.benchmark;

import java.util.Date;

import com.alibaba.excel.annotation.ExcelProperty;
import com.alibaba.excel.annotation.format.DateTimeFormat;

/** Java/Rust 共享性能契约中的四列表格行。 */
public class BenchmarkRow {

    @ExcelProperty(value = "ID", index = 0)
    private Long id;

    @ExcelProperty(value = "Name", index = 1)
    private String name;

    @ExcelProperty(value = "Date", index = 2)
    @DateTimeFormat("yyyy-MM-dd")
    private Date date;

    @ExcelProperty(value = "Score", index = 3)
    private Double score;

    public static BenchmarkRow fromId(long id) {
        BenchmarkRow row = new BenchmarkRow();
        row.setId(id);
        row.setName("row-" + id);
        row.setDate(new Date(1704067200000L + Math.floorMod(id, 28L) * 86_400_000L));
        row.setScore(id * 0.5d);
        return row;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public Double getScore() {
        return score;
    }

    public void setScore(Double score) {
        this.score = score;
    }
}
